lamindb/models/artifact.py

```diff
@@ -2815,6 +2816,22 @@ def delete(
         """
         super().delete(permanent=permanent, storage=storage, using_key=using_key)
 
+    @property
+    def _real_key(self) -> str | None:
+        """A real key for artifacts that represent paths in existing storages and also have a virtual key."""
```

Comment:
I'd make `_real_key` a public attribute called: `real_key`. Below is a tentative docstring. By having it public I think much transparency is created about the inner workings of lamindb for those interested.

An optional real storage key. 

It defaults to `f".lamindb/{uid}{suffix}"` if `_key_is_virtual = True` (`f".lamindb/{stem_uid}{suffix}"` if `overwrite_versions=True`). Otherwise it equals `key`.

If the user wants to deviate from this pattern, `_real_key` can be used to store the actual storage key.

Reply:
But still as an aux field, right?

Reply:
Now it is always written with the actual path.

Reply:
i thought it would be very confusing otherwise.

Reply:
I don't think we should duplicate the data. I think `_real_key` should be `None` in 95% of cases and only in those customized cases it's not `None`. I agree that the `uid`-based key is essentially immutable but it's still a lot of data duplication. 🤔

Reply:
I mean we haven't had a `_real_key` for 3 years now and it wasn't confusing and `None`. So adding now the optional custom key seems to be like a natural pattern for having something that's not `None` in rare cases.